py/server/deephaven/barrage.py

```diff
@@ -135,6 +136,7 @@ def barrage_session(host: str,
         tls_root_certs (bytes): PEM encoded root certificates to use for TLS connection, or None to use system defaults.
              If not None implies use a TLS connection and the use_tls argument should have been passed
              as True. Defaults to None
+        extra_headers (Dict[str, str]): extra headers to set when configuring the gRPC channel
 
```

Comment:
```suggestion
        extra_headers (Dict[str, str]): extra headers to set when configuring the gRPC channel. Defaults to None.

```

Reply:
Also are we sure the Core+ will not need client authentication soon? Even the answer is maybe, I think we might want to add that as well as it doesn't seem to be much more work and we can arrange the parameters in the same order as `pydeephaven.Session`.

Reply:
What do you mean by client authentication?

We use the auth_type and auth_token when creating a pydeephaven.session:
``` 
            super().__init__(host=host, port=port, auth_type=auth_type, auth_token=auth_token, use_tls=True,
                             extra_headers=extra_headers, tls_root_certs=session_manager.truststore_pem,
                             session_type=session_type,
                             **(session_arguments if session_arguments is not None else dict()))
```

Reply:
Server needs to authenticate the client or mutual TLS is what I mean.

Reply:
So @cpwright, do you think we might want mutual TLS in Core+ at some point? I.e. the client_cert_chain and client_private_key params (@jmao-denver will have to specify if this also includes client_opts or not).

Reply:
I do not expect we will be using mutual TLS soon; because we have alternative authentication methods and have not yet had a customer ask for it.

Reply:
@jmao-denver Whether it takes extra time is important because we require this fix in the upcoming (perhaps tomorrow) release.

To avoid rushing, should we just defer mutual TLS options?

Reply:
In trying to understand this from a birds-eye-view, I think Jianfeng is asking whether it would make sense to go ahead and add the rest of the stuff that is in Session that is missing from BarrageSession.

I think the answer to that would be "yes" at a high level.

I think Charles question on whether the testing for that would delay things makes sense, but I also think it is lacking some context.   I don't think we have ever tested those options (Client TLS) on the Session.  So testing them now for BarrageSession would be a higher standard of testing.   There is a question of consistency, and for me that's the angle I would personally use when approaching the question.  Session has arguments that try to support Client TLS.  It has never been tested.  BarrageSession has differences in its arguments and doesn't include that.  I think the differences in API between Session and BarrageSession are a problem of its own.   Certainly you can argue that both should be tested.  I would argue that either both exists or neither does.  So the decision seems to be "do we want to make them consistent and untested or do we want to remove the Session arguments for untested stuff."  My answer would be "make consistent and untested".  Why?  Because of the cost.  Since we are diving into the code now and Jianfeng is already staring at the code and making changes, and he has already stated that he doesn't think the addition of the code is much work.  So in my view his approach has the property of "no worse than before".

Reply:
@arman-ddl I am ready to approve as is. But since we do want to keep the two interfaces as close as possible and since @jcferretti added those extra parameters to the pydeephaven.Session constructor specifically for use by Core+, I just wanted to make sure that we don't miss something that Core+ maybe needs now. Based on Charles's comment and my conversation with @jcferretti about typical Core+ deployment and the reasons behind these extra parameters in the first place, I think it is OK we merge this and worry about what might be good to have but no needed right now in the future.